File: 2259. Remove Digit From Number to Maximize Result/solution.py
class Solution(object):
  def removeDigit(self, number, digit):
    """
    :type number: str
    :type digit: str
    :rtype: str
    """
    result = "0"
    for index, value in enumerate(number):
      if value == digit:
        temp = number[0:index] + number[index + 1:]
        if temp > result:
          result = temp
    return result

File: 2259. Remove Digit From Number to Maximize Result/solution.cpp
class Solution {
public:
  string removeDigit(string number, char digit) {
    string result = "0";
    for (int index = 0; index < number.size(); index++){
      char value = number[index];
      if (value == digit){
        string temp = number.substr(0, index) + number.substr(index + 1);
        if (temp > result){
          result = temp;
        }
      }
    }
    return result;
  }
};